tt_metal/impl/dispatch/kernels/cq_dispatch.cpp

```diff
@@ -789,6 +810,54 @@ static void process_delay_cmd() {
     cmd_ptr += sizeof(CQDispatchCmd);
 }
 
+FORCE_INLINE
+void process_go_signal_mcast_cmd() {
+    volatile CQDispatchCmd tt_l1_ptr *cmd = (volatile CQDispatchCmd tt_l1_ptr *)cmd_ptr;
+    volatile tt_l1_ptr uint32_t* worker_sem_addr = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(cmd->mcast.wait_addr);
+    aligned_go_signal = cmd->mcast.go_signal;
+    while (*worker_sem_addr < cmd->mcast.wait_count);
+    if (cmd->mcast.mcast_flag & GoSignalMcastSettings::SEND_MCAST) {
+        uint64_t dst = get_noc_addr_helper(worker_mcast_grid, mcast_go_signal_addr);
```

Comment:
this is a constexpr, right?

Reply:
Unfortunately not, this is a runtime check since we need to look at the dynamically configured command field.